src/lib/add-on/clients/add-on_auto.rb

```diff
@@ -1,328 +1,319 @@
-          # Adjust "name", bnc #434708
-          if srcid != nil && srcid != -1
-            repos = Pkg.SourceEditGet
-
-            found_at = -1
-            counter = -1
-
-            Builtins.foreach(repos) do |one_repo|
-              counter = Ops.add(counter, 1)
-              if Ops.get_integer(one_repo, "SrcId", -1) == srcid
-                found_at = counter
-                raise Break
-              end
-            end
-
-            if found_at != -1
-              name = Ops.get_string(repos, [found_at, "name"], "")
-
-              # Possibility to set name in control file, bnc #433981
-              if Builtins.haskey(prod, "name")
-                name = Ops.get_string(prod, "name", "")
-                Builtins.y2milestone("Preferred name: %1", name)
-                # Or use the one returned by Pkg::RepositoryScan
-              else
-                repos_at_url = Pkg.RepositoryScan(Pkg.ExpandedUrl(media))
-                # [ ["Product Name", "Path" ] ]
-                Builtins.foreach(repos_at_url) do |one_repo|
-                  if Ops.get(one_repo, 1, "") == pth
-                    name = Ops.get(one_repo, 0, "")
-                    raise Break
-                  end
-                end
-                Builtins.y2milestone("Preferred name: %1", name)
-              end
-
-              Ops.set(repos, [found_at, "name"], name)
-              Ops.set(repos, [found_at, "priority"], prod["priority"]) if prod.key?("priority")
-              Pkg.SourceEditSet(repos)
-            end
-          end
-          if Ops.get_string(prod, "product", "") != ""
-            Builtins.y2milestone(
-              "Installing product: %1",
-              Ops.get_string(prod, "product", "")
-            )
-            Pkg.ResolvableInstall(Ops.get_string(prod, "product", ""), :product)
-          else
-            Builtins.y2warning("No product to install")
-          end
-        end
 
-        # reread agents, redraw wizard steps, etc.
-        AddOnProduct.ReIntegrateFromScratch
-
-        @ret = true
-      # Reads configuration of add-ons from the current system
-      # to memory. To get that configuration, use Export() functionality.
-      #
-      # @return [Boolean]
-      elsif @func == "Read"
-        if !PackageLock.Check
-          Builtins.y2error("Cannot get package lock")
-          return false
+      AddOnProduct.Import("add_on_products" => valid_add_on_products)
+    end
+
+    def summary
+      formatted_add_on = AddOnProduct.add_on_products.map do |add_on|
+        "<li>" \
+          "Media: #{add_on["media_url"]}, " \
+          "Path: #{add_on["product_dir"]}, " \
+          "Product: #{add_on["product"]}" \
+        "</li>"
+      end
+
+      ["<ul>", formatted_add_on, "</ul>"].flatten.join("\n")
+    end
+
+    def modified?
+      AddOnProduct.modified
+    end
+
+    def modified
+      AddOnProduct.modified = true
+    end
+
+    def reset
+      AddOnProduct.add_on_products = []
+
+      {}
```

Comment:
not needed. Reset return value should not be used see https://github.com/yast/yast-yast2/blob/0d1f6001e99734c8da163d78c04bf53b029ba8c4/library/general/src/lib/installation/auto_client.rb#L114